### PR TITLE
update gateway api, gateway can enable ravenl3/ravenl7 and can select…

### PR DESCRIPTION
**Pull request by River-sh**
… multiple endpoints for ravenl7

> /kind enhancement
> /kind feature
> /sig network

#### What this PR does / why we need it:
1. support to enable/unenable raven l7 proxy 
2. support to enable/unenable raven l3 tunnel
3. support for electing multiple replicas
4. support watch lb service for update the public ip for exposed gateway

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/apis/raven/v1alpha1/gateway_types.go [hunk 1]
```diff
@@ -32,60 +29,94 @@ const (
...
+	// ActiveEndpoints is the reference of the active endpoint.
+	ActiveEndpoints []*Endpoint `json:"activeEndpoints,omitempty"`
 }
 
 // +genclient
 // +k8s:openapi-gen=true
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:scope=Cluster,path=gateways,shortName=gw,categories=all
-//+kubebuilder:printcolumn:name="ActiveEndpoint",type=string,JSONPath=`.status.activeEndpoint.nodeName`
```
Comment (njucjc) [inline pkg/apis/raven/v1alpha1/gateway_types.go:88]: without this line, `ActiveEndponts` will not display when you get it by `kubectl`

**Comment by luckymrwang** [comment]
@River-sh The gateway CRD has been modified some fields and type. And does it support older version not a new cluster?

**Comment by River-sh** [comment]
@luckymrwang  The new gateway crd can support the old version, which only adds the Spec attribute

File: pkg/apis/raven/v1alpha1/gateway_types.go (hunk 1 above)
Comment (River-sh) [reply pkg/apis/raven/v1alpha1/gateway_types.go:88]: The gateway supports multiple ActiveEndpoints, therefore, how to display them is not considered for the time being

**Pull request closed without merging**